spec.html

```diff
@@ -36557,27 +36557,73 @@
+                1. Insert _d_ as the first element of _functionsToInitialize_.
+          1. For each element _d_ in _functionsToInitialize_ do
+            1. Let _fn_ be the sole element of the BoundNames of _d_.
+            1. Let _fo_ be the result of performing InstantiateFunctionObject for _d_ with argument _env_.
+            1. Perform _envRec_.InitializeBinding(_fn_, _fo_).
+        </emu-alg>
+      </emu-annex>
       <emu-annex id="sec-web-compat-functiondeclarationinstantiation">
         <h1>Changes to FunctionDeclarationInstantiation</h1>
         <p>During FunctionDeclarationInstantiation (<emu-xref href="#sec-functiondeclarationinstantiation"></emu-xref>) the following steps are performed in place of step 29:</p>
         <emu-alg>
           1. If _strict_ is *false*, then
-            1. For each |FunctionDeclaration| _f_ that is directly contained in the |StatementList| of a |Block|, |CaseClause|, or |DefaultClause|,
-              1. Let _F_ be StringValue of the |BindingIdentifier| of |FunctionDeclaration| _f_.
-              1. If replacing the |FunctionDeclaration| _f_ with a |VariableStatement| that has _F_ as a |BindingIdentifier| would not produce any Early Errors for _func_ and _F_ is not an element of BoundNames of _argumentsList_, then
-                1. NOTE A var binding for _F_ is only instantiated here if it is neither a VarDeclaredName, the name of a formal parameter, or another |FunctionDeclaration|.
-                1. If _instantiatedVarNames_ does not contain _F_, then
-                  1. Perform ! _varEnvRec_.CreateMutableBinding(_F_).
-                  1. Perform _varEnvRec_.InitializeBinding(_F_, *undefined*).
-                  1. Append _F_ to _instantiatedVarNames_.
-                1. When the |FunctionDeclaration| _f_ is evaluated, perform the following steps in place of the |FunctionDeclaration| Evaluation algorithm provided in <emu-xref href="#sec-function-definitions-runtime-semantics-evaluation"></emu-xref>:
-                  1. Let _fenv_ be the running execution context's VariableEnvironment.
-                  1. Let _fenvRec_ be _fenv_'s EnvironmentRecord.
-                  1. Let _benv_ be the running execution context's LexicalEnvironment.
-                  1. Let _benvRec_ be _benv_'s EnvironmentRecord.
-                  1. Let _fobj_ be ! _benvRec_.GetBindingValue(_F_, *false*).
-                  1. Perform ! _fenvRec_.SetMutableBinding(_F_, _fobj_, *false*).
-                  1. Return NormalCompletion(~empty~).
+            1. For each |Block|, |CaseClause|, or |DefaultClause| _block_,
+              1. Let _functionsInBlock_ be an empty List.
+              1. For each |FunctionDeclaration| _f_ that is directly contained in the |StatementList| of _block_, in reverse order do
+                1. If _f_ is not an element of _functionsInBlock_, then
```

Comment:
This condition is always true, you need to add a separate list of function names to keep track of duplicates.